### PR TITLE
Extract display name and description from JavaDoc

### PR DESCRIPTION
**Pull request by timtebeek**
For https://github.com/PicnicSupermarket/error-prone-support/pull/925 & @Stephan202

### CODE REVIEW

File: src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java [hunk 1]
```diff
@@ -399,6 +400,27 @@ private String recipeDescriptor(JCTree.JCClassDecl classDecl, String defaultDisp
                 String displayName = defaultDisplayName;
                 String description = defaultDescription;
                 Set<String> tags = new LinkedHashSet<>();
+
+                // Extract from JavaDoc
+                Tokens.Comment comment = cu.docComments.getComment(classDecl);
+                if (comment != null && comment.getText() != null && !comment.getText().isEmpty()) {
+                    String commentText = comment.getText()
+                            .replace("{@link ", "`")
+                            .replace("}", "`")
```
Comment (Stephan202) [inline src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java:409]: I guess `{@code` should get the same treatment.

(This wouldn't work for `{@value`, but it seems highly unlikely one would use that for a Refaster rule.)

**Review by Stephan202 — COMMENTED** [review]
Nice! Within Error Prone Support we currently have [similar functionality](https://github.com/PicnicSupermarket/error-prone-support/blob/7529b9925145c3fdaf0864a6df99b6cbb20d7d7f/error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java#L390-L391) using an `@Description` annotation, but I have pondered in the past whether indeed we should instead just look at the Javadoc. So this change SGTM!

**Comment by timtebeek** [comment]
> Nice! Within Error Prone Support we currently have [similar functionality](https://github.com/PicnicSupermarket/error-prone-support/blob/7529b9925145c3fdaf0864a6df99b6cbb20d7d7f/error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java#L390-L391) using an `@Description` annotation, but I have pondered in the past whether indeed we should instead just look at the Javadoc. So this change SGTM!

Ah interesting! Had been looking for something like that on your end, but didn't immediately spot anything. I only see that `@tech.picnic.errorprone.refaster.annotation.Description` [applied twice](https://github.com/search?q=repo%3APicnicSupermarket%2Ferror-prone-support+%22%40Description%22&type=code) though, once in tests. I'm guessing you're considering just taking the JavaDoc there as well then?

File: src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java (hunk 1 above)
Comment (timtebeek) [reply src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java]: Thanks! Picked up in https://github.com/openrewrite/rewrite-templating/pull/51/commits/fc3bb8d87c57c47fd35306d36588d1b231880ddf to replace any such tags.

**Comment by Stephan202** [comment]
> I'm guessing you're considering just taking the JavaDoc there as well then?

Nothing decided, but at this point it seems likely: the JavaDoc seems more maintainable and easy to use, especially when considering line wrapping.

**Pull request merged**